src/protocol/requests/joinGroup/v4/response.js

```diff
@@ -0,0 +1,33 @@
+const { decode } = require('../v3/response')
+const { failure, createErrorFromCode } = require('../../../error')
+
+/**
+ * JoinGroup Response (Version: 4) => throttle_time_ms error_code generation_id group_protocol leader_id member_id [members]
+ *   throttle_time_ms => INT32
+ *   error_code => INT16
+ *   generation_id => INT32
+ *   group_protocol => STRING
+ *   leader_id => STRING
+ *   member_id => STRING
+ *   members => member_id member_metadata
+ *     member_id => STRING
+ *     member_metadata => BYTES
+ */
+
+const parse = async data => {
+  if (failure(data.errorCode)) {
+    const error = createErrorFromCode(data.errorCode)
+    if (error.type === 'MEMBER_ID_REQUIRED') {
+      error.memberId = data.memberId
```

Comment:
This looks like the cleanest solution, i was also thinking on changing inside `createErrorFromCode` function but that function does not get full data set and will require updating every single parser or passing second param which looks ugly to only support single case

Reply:
Right, I didn't think about the fact that the protocol error doesn't include the response. Take a look at what we're doing with `KafkaJSOffsetOutOfRange`, which has a similar problem. I think that solution is appropriate here as well. Basically create a new `MemberIdRequired` class that extends `KafkaJSProtocolError` and adds a a `memberId` field, and then throw that instead of a generic protocol error if the type is `MEMBER_ID_REQUIRED`.

Reply:
Eventually got some time, have updated PR based on above !